### PR TITLE
Add INN Generator

### LINKED ISSUE #1004: Add support for Hibernate Validator `@INN` annotation
### Feature description

Checks that the annotated character sequence represents a Russian taxpayer identification number. Can be applied to both individual and juridical versions of INN

-  Supported data types: `CharSequence`
- Country:  Russia

Docs: https://docs.jboss.org/hibernate/stable/validator/reference/en-US/html_single/#_country_specific_constraints

`@REGON` implementation can be used for reference (`RegonGeneratorSpec` and `RegonSpec`).

### PR DESCRIPTION
**Pull request by EvaristeGalois11**
Hi, I finally found the time to finish this PR :D

There isn't much info about the INN specification and I unfortunately don't speak Russian, so I just reverse engineered what Hibernate is doing which is only a formal check on the digits. It should be enough to comply with the validation check.

Close #1004

### CODE REVIEW

File: instancio-core/src/main/java/org/instancio/generator/specs/rus/InnGeneratorSpec.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.instancio.generator.specs.rus;
+
+import org.instancio.generator.specs.NullableGeneratorSpec;
+
+/**
+ * Spec for generating <a href="https://ru.wikipedia.org/wiki/%D0%98%D0%B4%D0%B5%D0%BD%D1%82%D0%B8%D1%84%D0%B8%D0%BA%D0%B0%D1%86%D0%B8%D0%BE%D0%BD%D0%BD%D1%8B%D0%B9_%D0%BD%D0%BE%D0%BC%D0%B5%D1%80_%D0%BD%D0%B0%D0%BB%D0%BE%D0%B3%D0%BE%D0%BF%D0%BB%D0%B0%D1%82%D0%B5%D0%BB%D1%8C%D1%89%D0%B8%D0%BA%D0%B0">
```
Comment (armandino) [inline instancio-core/src/main/java/org/instancio/generator/specs/rus/InnGeneratorSpec.java:21]:
```suggestion
 * Spec for generating <a href="https://w.wiki/9iyP">
```

File: instancio-core/src/main/java/org/instancio/generator/specs/rus/InnGeneratorSpec.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * Russian taxpayer identification number' </a> (INN).
```
Comment (armandino) [inline instancio-core/src/main/java/org/instancio/generator/specs/rus/InnGeneratorSpec.java:22]:
```suggestion
 * Russian taxpayer identification number'</a> (INN).
```

File: instancio-core/src/main/java/org/instancio/generator/specs/rus/InnSpec.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.instancio.generator.specs.rus;
+
+import org.instancio.generator.ValueSpec;
+
+/**
+ * Spec for generating <a href="https://ru.wikipedia.org/wiki/%D0%98%D0%B4%D0%B5%D0%BD%D1%82%D0%B8%D1%84%D0%B8%D0%BA%D0%B0%D1%86%D0%B8%D0%BE%D0%BD%D0%BD%D1%8B%D0%B9_%D0%BD%D0%BE%D0%BC%D0%B5%D1%80_%D0%BD%D0%B0%D0%BB%D0%BE%D0%B3%D0%BE%D0%BF%D0%BB%D0%B0%D1%82%D0%B5%D0%BB%D1%8C%D1%89%D0%B8%D0%BA%D0%B0">
```
Comment (armandino) [inline instancio-core/src/main/java/org/instancio/generator/specs/rus/InnSpec.java:21]:
```suggestion
 * Spec for generating <a href="https://w.wiki/9iyP">
```

File: instancio-core/src/main/java/org/instancio/generator/specs/rus/InnSpec.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ * Russian taxpayer identification number' </a> (INN).
```
Comment (armandino) [inline instancio-core/src/main/java/org/instancio/generator/specs/rus/InnSpec.java:22]:
```suggestion
 * Russian taxpayer identification number'</a> (INN).
```

**Comment by EvaristeGalois11** [comment]
It seems that the Sonar workflow is broken... again... :tada: 

I'll try to look into it, but it seems that the tar command doesn't find the .m2/repository directory anymore. That's odd considering that it was working fine 10 hours ago. :thinking:

**Review by armandino — COMMENTED** [review]
@EvaristeGalois11 many thanks for the PR! Looks good to me. I just left a couple of javadoc suggestions to make the URL a bit shorter using https://en.wikipedia.org/wiki/Special:UrlShortener.

File: instancio-core/src/main/java/org/instancio/generator/specs/rus/InnGeneratorSpec.java (hunk 1 above)
Comment (EvaristeGalois11) [reply instancio-core/src/main/java/org/instancio/generator/specs/rus/InnGeneratorSpec.java:21]: Oh I didn't know that wikipedia had an url shortener, that's cool!

**Review by armandino — APPROVED** [review]

**Comment by EvaristeGalois11** [comment]
Thank you @armandino for the review, I applied your suggestions.

> I recall it happened once in the past and then resolved by itself after a while.

Mmmm that's strange, it could be that the github worker sometimes setups the workspace in a slightly different way. Anyway if the problem is just the missing directory it should be easy to fix. I'll try to release a new version of the workflow asap.

**Comment by armandino** [comment]
> There isn't much info about the INN specification and I unfortunately don't speak Russian, so I just reverse engineered what Hibernate is doing which is only a formal check on the digits.

That's amazing :smile: 

Thanks for updating the PR!

**Comment by armandino** [comment]
> Mmmm that's strange, it could be that the github worker sometimes setups the workspace in a slightly different way. Anyway if the problem is just the missing directory it should be easy to fix. I'll try to release a new version of the workflow asap.

Thanks for looking into it!

**Pull request merged**